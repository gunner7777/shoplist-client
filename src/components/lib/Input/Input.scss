.Input {
  margin: 15px 0;

  &-Input {
    width: 100%;
    padding: 4px;
    height: 44px;
    border: 1px solid #000;
  }

  &-Label {
    display: block;
    margin-bottom: 4px;
  }
}