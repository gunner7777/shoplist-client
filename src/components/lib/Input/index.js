import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './Input.scss';

class Input extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      prevProps: '',
      value: '',
    };

    this.handleChange = this.handleChange.bind(this);
  }

  static getDerivedStateFromProps(props, state) {
    const prevProps = state.prevProps;
    const currentValue = props.currValue !== prevProps ? props.currValue : state.value;

    return {
      prevProps: props.currValue,
      value: currentValue,
    };
  }

  handleChange(event) {
    this.setState({
      value: event.target.value,
    });
  }

  render() {
    const { label } = this.props;

    return (
      <div className="Input">
        <label className="Input-Block">
          <span className="Input-Label">{label}</span>
          <input
            className="Input-Input"
            type="text"
            onChange={this.handleChange}
            value={this.state.value}
            placeholder={label || 'enter value'}
          />
        </label>
      </div>
    );
  }
}

Input.propTypes = {
  label: PropTypes.string,
};

export default Input;
