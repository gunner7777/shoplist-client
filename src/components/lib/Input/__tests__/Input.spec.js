import React from 'react';
import { shallow } from 'enzyme';
import Input from '../index';

describe('input tests', () => {
  describe('input rendering', () => {
    it('input initial rendering', () => {
      const InputComponent = shallow(<Input />);
      expect(InputComponent).toMatchSnapshot();
    });
  });

  describe('props and state tests', () => {
    const initialState = {
      prevProps: '',
      value: '',
    };

    const InputComponent = shallow(<Input label="text" />);

    it('label for input', () => {
      // const InputComponent = shallow(<Input label="text" />);
      expect(InputComponent.find('.Input-Label').text()).toEqual('text');
    });

    it('if change text in input', () => {
      const newText = '111';
      // const InputComponent = shallow(<Input label="text" />);
      InputComponent.find('.Input-Input').simulate('change', {
        target: { value: newText },
      });
      expect(InputComponent.state().value).toEqual(newText);
    });

    afterAll(() => {
      InputComponent.setState(initialState);
    });
  });
});
