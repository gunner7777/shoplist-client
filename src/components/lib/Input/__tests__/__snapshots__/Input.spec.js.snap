// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input tests input rendering input initial rendering 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    label="enter"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        className="Input-Block"
      >
        <span
          className="Input-Label"
        >
          enter
        </span>
        <input
          className="Input-Input"
          onChange={[Function]}
          placeholder="enter"
          type="text"
        />
      </label>,
      "className": "Input",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="Input-Label"
          >
            enter
          </span>,
          <input
            className="Input-Input"
            onChange={[Function]}
            placeholder="enter"
            type="text"
          />,
        ],
        "className": "Input-Block",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "enter",
            "className": "Input-Label",
          },
          "ref": null,
          "rendered": "enter",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "Input-Input",
            "onChange": [Function],
            "placeholder": "enter",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          className="Input-Block"
        >
          <span
            className="Input-Label"
          >
            enter
          </span>
          <input
            className="Input-Input"
            onChange={[Function]}
            placeholder="enter"
            type="text"
          />
        </label>,
        "className": "Input",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="Input-Label"
            >
              enter
            </span>,
            <input
              className="Input-Input"
              onChange={[Function]}
              placeholder="enter"
              type="text"
            />,
          ],
          "className": "Input-Block",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "enter",
              "className": "Input-Label",
            },
            "ref": null,
            "rendered": "enter",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "Input-Input",
              "onChange": [Function],
              "placeholder": "enter",
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
